diff --git a/drivers/base/dd.c b/drivers/base/dd.c
index 0c3725c3e..e53c2ce9d 100644
--- a/drivers/base/dd.c
+++ b/drivers/base/dd.c
@@ -28,6 +28,8 @@
 #include <linux/pm_runtime.h>
 #include <linux/pinctrl/devinfo.h>
 #include <linux/slab.h>
+#include <linux/of.h>
+#include <linux/kernel.h>
 
 #include "base.h"
 #include "power/power.h"
@@ -1169,6 +1171,9 @@ static int __driver_attach(struct device *dev, void *data)
 	 * is an error.
 	 */
 
+	if(dev->of_node){
+		printk("dtb node: %s\n", dev->of_node->name);
+	}
 	ret = driver_match_device(drv, dev);
 	if (ret == 0) {
 		/* no match */
